src/renderers/shaders/ShaderLib/backgroundCube.glsl.js

```diff
@@ -0,0 +1,50 @@
+export const vertex = /* glsl */`
+varying vec3 vWorldDirection;
+
+#include <common>
+
+void main() {
+
+	vWorldDirection = transformDirection( position, modelMatrix );
+
+	#include <begin_vertex>
+	#include <project_vertex>
+
+	gl_Position.z = gl_Position.w; // set z to camera.far
+
+}
+`;
+
+export const fragment = /* glsl */`
+#include <envmap_common_pars_fragment>
+uniform float backgroundBlurriness;
+uniform float opacity;
+
+varying vec3 vWorldDirection;
+
+#include <cube_uv_reflection_fragment>
+
+void main() {
+
+	#ifdef ENVMAP_TYPE_CUBE
+
+		vec4 envColor = textureCube( envMap, vec3( flipEnvMap * vWorldDirection.x, vWorldDirection.yz ) );
+
+	#elif defined( ENVMAP_TYPE_CUBE_UV )
+
+		vec4 envColor = textureCubeUV( envMap, vWorldDirection, backgroundBlurriness );
+
+	#else
+
+		vec4 envColor = vec4( 0.0 );
+
+	#endif
+
+	gl_FragColor = envColor;
+	gl_FragColor.a *= opacity;
```

Comment:
@Mugen87 Is opacity support needed here?

Reply:
I would say no. We should consider background definitions as opaque.

Reply:
OK, removed!